amaranth/build/run.py

```diff
@@ -92,23 +87,59 @@ def execute_local(self, root="build", *, run_script=True, env=None):
                     content = content.encode("utf-8")
                 with open(filename, "wb") as f:
                     f.write(content)
-
-            if run_script:
-                if sys.platform.startswith("win32"):
-                    # Without "call", "cmd /c {}.bat" will return 0.
-                    # See https://stackoverflow.com/a/30736987 for a detailed explanation of why.
-                    # Running the script manually from a command prompt is unaffected.
-                    subprocess.check_call(["cmd", "/c", f"call {self.script}.bat"],
-                                          env=os.environ if env is None else env)
-                else:
-                    subprocess.check_call(["sh", f"{self.script}.sh"],
-                                          env=os.environ if env is None else env)
-
-            return LocalBuildProducts(os.getcwd())
-
+            return os.getcwd()
         finally:
             os.chdir(cwd)
 
+    def execute_local(self, root="build", *, run_script=True, env=None):
+        """
+        Execute build plan using the local strategy. Files from the build plan are placed in
+        the build root directory ``root``, and, if ``run_script`` is ``True``, the script
+        appropriate for the platform (``{script}.bat`` on Windows, ``{script}.sh`` elsewhere)
+        is executed in the build root. If ``env`` is not ``None``, the environment is replaced
+        with ``env``.
+
+        Returns :class:`LocalBuildProducts`.
+        """
+        build_dir = self.extract(root)
+        if run_script:
+            if sys.platform.startswith("win32"):
+                # Without "call", "cmd /c {}.bat" will return 0.
+                # See https://stackoverflow.com/a/30736987 for a detailed explanation of why.
+                # Running the script manually from a command prompt is unaffected.
+                subprocess.check_call(["cmd", "/c", f"call {self.script}.bat"],
+                                        env=os.environ if env is None else env)
+            else:
+                subprocess.check_call(["sh", f"{self.script}.sh"],
+                                        env=os.environ if env is None else env)
+
+        return LocalBuildProducts(build_dir)
+
+
+    def execute_local_docker(self, root="build", *, container_image, container_workdir="/build",
```

Comment:
```suggestion
    def execute_local_docker(self, root="build", *, image, mountpoint="/build",
```

I'd also probably just drop the ability to customize the mounpoint.